doc/spec/skupper-cli-spec.md

```diff
@@ -0,0 +1,287 @@
+Waiting for site "west" to delete...
+Site "west" is deleted
+~~~
+
+## Example token operations
+
+~~~ console
+$ skupper token create token.yaml
+Token file created at token.yaml
+The token expires after 1 use or after 15 minutes
+~~~
+
+## Example link operations
+
+~~~ console
+$ skupper link create token.yaml
+Waiting for link "west" to become active...
+Link "west" is active
+You can now delete token.yaml
+
+$ skupper link get
+NAME    STATUS   COST
+south   Error    10
+west    Active   1
+
+$ skupper link get west
+NAME   STATUS   COST
+west   Active   1
+
+$ skupper link get west -o yaml
+apiVersion: v2alpha1
+kind: Link
+[...]
+
+$ skupper link delete west
+Waiting for link "west" to delete...
+Link "west" is deleted
+~~~
+
+## Example listener operations
+
+~~~ console
+$ skupper listener create database --host database --port 5432
+Waiting for listener...
+Listener "database" is ready
+
+$ skupper listener get
+NAME       ROUTING-KEY   HOST       PORT
+payments   payments      payments   8080
+database   database      database   5432
+
+$ skupper listener get database -o yaml
+apiVersion: v2alpha1
+kind: Listener
+[...]
+
+$ skupper listener set database --port 5431
+Waiting for listener...
+Listener "database" is ready
+
+$ skupper listener delete database
+Listener "database" is deleted
+~~~
+
+## Example connector operations
+
+~~~ console
+$ skupper connector create database --workload deployment/database --port 5432
+Waiting for connector...
+Connector "database" is ready
+
+$ skupper connector get
+NAME         ROUTING-KEY   SELECTOR       HOST   PORT
+database     database      app=database   -      5432
+
+$ skupper connector get database -o yaml
+apiVersion: v2alpha1
+kind: Connector
+[...]
+
+$ skupper connector set database --port 5431
+Waiting for connector...
+Connector "database" is ready
+
+$ skupper connector delete database
+Connector "database" is deleted
+~~~
+
+## Skupper resource commands
+
+These are the core Skupper commands.  They are not the only commands,
+however.  Additional commands will be added to the spec in the future.
+For instance, we will have commands for debugging, getting versions,
+and for installing Skupper on a platform.
+
+### `skupper <resource-type>`
+
+Print help text for the operations of this resource type.
+
+### `skupper <resource-type> create [options]`
+
+Create a resource.
+
+Resource options are set using one or more `--some-key some-value`
+command line options.  YAML resource options in camel case (`someKey`)
+are exposed as hyphenated names (`some-key`) when used as command line
+options.
+
+#### Positional arguments
+
+`token create <token-file>` - `token create` takes one positional
+parameter, the file in which to create the token.
+
+`link create <token-file>` - `link create` takes one positional
+parameter, the file containing the token.
+
+`listener create <routing-key>` - `listener create` takes one
+positional parameter, the routing key used to match the listener to
```

Comment:
To me this doesn't feel intuitive or consistent. The positional parameter should be the name of the resource being created. We can of course state that the --routing-key option if not specified defaults to the name of the resource.